sink_service/source/config.c

```diff
@@ -776,6 +776,129 @@ static int get_config_data_content(sd_bus_message *m, void *userdata, sd_bus_err
     return sd_bus_send(sd_bus_message_get_bus(reply), reply, NULL);
 }
 
+/**
+ * \brief Read security keys from the message
+ *
+ * Message is expected to have two byte arrays for cipher and authentication
+ * keys, and a key sequence.
+ *
+ * \param[in,out] message
+ *                Message to read the keys from
+ * \param[out]    error
+ *                Pointer to the sd_bus error of the request message
+ * \param[out]    security_keys
+ *                Address to write the security keys
+ * \return  On success, a non-negative value. On failure, a negative errno-style
+ *          code consistent with other sd_bus methods.
+ */
+static int read_security_keys_from_message(sd_bus_message *const message,
+                                           sd_bus_error *const error,
+                                           wpc_key_pair_t *const security_keys)
+{
+    const void *key_bytes;
+    size_t key_size;
+
+    int r = sd_bus_message_read_array(message, 'y', &key_bytes, &key_size);
+    if (r < 0)
+    {
+        sd_bus_error_set_errno(error, r);
+        LOGE("Could not read the cipher key: %s\n", strerror(-r));
+        return r;
+    }
+    if (key_size != sizeof(security_keys->key_pair.encryption))
+    {
+        const app_res_e wpc_res = APP_RES_INVALID_VALUE;
+        SET_WPC_ERROR(error, __FUNCTION__, wpc_res);
+        LOGE("Invalid cipher key size (%zu) (ret=%d)\n", key_size, wpc_res);
+        return -EINVAL;
+    }
+
+    memcpy((void*)security_keys->key_pair.encryption, key_bytes, sizeof(security_keys->key_pair.encryption));
+
+    r = sd_bus_message_read_array(message, 'y', &key_bytes, &key_size);
+    if (r < 0)
+    {
+        sd_bus_error_set_errno(error, r);
+        LOGE("Could not read the authentication key: %s\n", strerror(-r));
+        return r;
+    }
+    if (key_size != sizeof(security_keys->key_pair.authentication))
+    {
+        const app_res_e wpc_res = APP_RES_INVALID_VALUE;
+        SET_WPC_ERROR(error, __FUNCTION__, wpc_res);
+        LOGE("Invalid authentication key size (%zu) (ret=%d)\n", key_size, wpc_res);
+        return -EINVAL;
+    }
+
+    memcpy((void*)security_keys->key_pair.authentication, key_bytes, sizeof(security_keys->key_pair.authentication));
+
+    r = sd_bus_message_read(message, "y", &security_keys->sequence_number);
+    if (r < 0)
```

Comment:
Should we also here validate that the sequence falls into the supported range `[1..255]`? Or does we let sink node itself to handle that (or will the APIs just silently crop the value)?